sweep/sweeper.go

```diff
@@ -1319,30 +1323,82 @@ func (s *UtxoSweeper) ListSweeps() ([]chainhash.Hash, error) {
 	return s.cfg.Store.ListSweeps()
 }
 
+// mempoolLookup takes an input's outpoint and queries the mempool to see
+// whether it's already been spent in a transaction found in the mempool.
+// Returns the transaction if found.
+func (s *UtxoSweeper) mempoolLookup(op wire.OutPoint) (*wire.MsgTx, bool) {
+	// For neutrino backend, there's no mempool available, so we exit
+	// early.
+	if s.cfg.Mempool == nil {
+		log.Debugf("Skipping mempool lookup for %v, no mempool ", op)
+
+		return nil, false
+	}
+
+	// Make a subscription to the mempool. If this outpoint is already
+	// spent in mempool, we should get a spending event back immediately.
+	mempoolSpent, err := s.cfg.Mempool.SubscribeMempoolSpent(op)
+	if err != nil {
+		log.Errorf("Unable to subscribe to mempool spend for input "+
+			"%v: %v", op, err)
+
+		return nil, false
+	}
+
+	// We want to cancel this subscription in the end as we are only
+	// interested in a one-time query and this subscription won't be
+	// listened once this method returns.
+	defer s.cfg.Mempool.CancelMempoolSpendEvent(mempoolSpent)
+
+	// Do a non-blocking read on the spent event channel.
+	select {
+	case details := <-mempoolSpent.Spend:
```

Comment:
Only if this outpoint has already been spent in mempool, so it will block when a) the tx is not in mempool, or b) the tx has already been confirmed, which are both fine as we are doing a non-blocking read here.

Reply:
Sure, so here are the lines, 
1. `SubscribeMempoolSpent` calls `NotifySpend` here:
https://github.com/lightningnetwork/lnd/blob/d41b86bb7e64885703cb25a2e29d9442dcfe3b99/chainntnfs/bitcoindnotify/bitcoind.go#L1060
2. `NotifySpend` then calls `LookupInputSpend` [here](https://github.com/btcsuite/btcwallet/blob/1f3534b00d1440cd6c7f4b68e7dedeeed51d3d84/chain/bitcoind_client.go#L271)
3. `LookupInputSpend` will return the tx if it's already in mempool, as validated by this [test](https://github.com/btcsuite/btcwallet/blob/master/chain/bitcoind_events_test.go#L169).

Reply:
That doesn't look immediate because the call to `onRelevantTx` puts the tx into a concurrent queue

Reply:
Turns out you are right. This leads to a bigger change as there's no way to do mempool look ups for `btcd`, and for `bitcoind` it's easier as we already have the method in place. So we'll need to 
1. implement `gettxspendingprevout` in `btcd`, https://github.com/btcsuite/btcd/pull/2125
2. add a mempool lookup method in `btcwallet`, https://github.com/btcsuite/btcwallet/pull/910
3. make use of it here.